nexus/src/db/model/service.rs

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::ServiceKind;
+use crate::db::ipv6;
+use crate::db::schema::service;
+use db_macros::Asset;
+use std::net::Ipv6Addr;
+use uuid::Uuid;
+
+/// Representation of services which may run on Sleds.
+#[derive(Queryable, Insertable, Debug, Clone, Selectable, Asset)]
+#[diesel(table_name = service)]
+pub struct Service {
+    #[diesel(embed)]
+    identity: ServiceIdentity,
+
+    pub sled_id: Uuid,
+    pub ip: ipv6::Ipv6Addr,
```

Comment:
Do these objects need to track their port? I'm not sure it matters for identifying a service running on a sled, e.g., to know `nexus` is on a sled we only need the IP, and not the port. But how does that information get propagated out to DNS, for example?

Reply:
Frankly Nexus was the big one that drove me towards excluding it, but based on our conversation yesterday, that may be in flux.

Basically, I wanted to support services which provided different APIs on different ports. On main, Nexus services the internal and external APIs on the same IP, but different ports.

I figured the port could be derived based on the `ServiceKind`.

Now, for the case of Nexus specifically, I think the internal/external interfaces will actually end up on different IPs - but it's not clear if we want to add the constraint that a "single service can only listen on a single port" in perpetuity.